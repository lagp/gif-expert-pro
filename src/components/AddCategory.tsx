import PropTypes from 'prop-types'
import { ChangeEvent, FormEvent, useState } from 'react'

interface AddCategoryProps {
  onNewCategory: (category: string) => void
}

export const AddCategory = ({ onNewCategory }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
    setInputValue(target.value)
  }

  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    if (inputValue.trim().length <= 1) {
      return
    }

    onNewCategory(inputValue.trim())
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input type='text' placeholder='Buscar gifs' value={inputValue} onChange={onInputChange} />
    </form>
  )
}

AddCategory.prototype = {
  onNewCategory: PropTypes.func.isRequired,
}
