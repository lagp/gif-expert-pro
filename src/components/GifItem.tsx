import PropTypes from 'prop-types'

export interface GifItemProps {
  title: string
  url: string
}

export const GifItem = ({ title, url }: GifItemProps) => {
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
