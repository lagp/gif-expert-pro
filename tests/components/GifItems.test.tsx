import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('GifItems testing', () => {
  const title = 'Saitama'
  const url =
    'https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=b042654ee79e4bs4kw3qad01n21i9krofgspxw1doc571a9d&ep=v1_gifs_search&rid=giphy.gif&ct=g'

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })

  test('should show the image with the indicated url and alt property', () => {
    render(<GifItem title={title} url={url} />)

    const { src, alt } = screen.getByRole('img') as HTMLImageElement

    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should show the title in the component', () => {
    render(<GifItem title={title} url={url} />)

    expect(screen.getByText(title)).toBeTruthy()
  })
})
