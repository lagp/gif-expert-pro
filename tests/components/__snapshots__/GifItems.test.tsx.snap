// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItems testing should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=b042654ee79e4bs4kw3qad01n21i9krofgspxw1doc571a9d&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
